modules/jdbc/src/main/java/org/testcontainers/jdbc/ContainerDatabaseDriver.java

```diff
@@ -199,7 +203,7 @@ private Connection wrapConnection(final Connection connection, final JdbcDatabas
 
         return new ConnectionWrapper(connection, () -> {
             finalConnections.remove(connection);
-            if (finalConnections.isEmpty()) {
+            if (!isDaemon && finalConnections.isEmpty()) {
```

Comment:
This line only prevents running container from stopping if `TC_DAEMON=true` was specified in connection string.

`ContainerDatabaseDriver ` maintains a mapping `jdbcUrlContainerCache` which maps JDBC connection string to the container instance. New container won't start If such a mapping exists.

My change basically ensures that this mapping is **not** removed once **all** connections are closed.